### PR TITLE
redo demo for authoring

### PR DESCRIPTION
**Pull request by richfirely**
1. Merged all the great code from CLI.CMS to CLI and retired CLI.CMS
2. Added Measures.Authoring project for creating new CQL measures.
3. Made all the measures projects exe's for simplicity
4. Removed CLI direct dependency on Measures projects. Goes the other way around.
5. Retired graph logic and loop through resources directly
6. Added valueset checking to CLI
7. Added test data and related input directories to align with vscode tooling.
More...

### CODE REVIEW

**Comment by baseTwo** [comment]
CLI input argument validations should not be displayed as an exception to the end-user. (perhaps another issue for later if you are pressed for time)

**Comment by baseTwo** [comment]
Runtime exceptions when  running CLI:
* BCSE profile: System.ArgumentException: Please choose one: either assemblyPath or resourcesDirectory.  StackTrace: CommandLineOptions.cs:line 62
* AdultAO_FromResources and AdultAO_FromProject profiles:  System.ArgumentException: The value (options.OutputFile) cannot be an empty string. LibraryRunner 31

I suspect it's because the argument names and values that can be provided in the new CommandLineOptions have not been applied to launchSettings.json

File: Demo/CLI/Helpers/CommandLineOptions.cs [hunk 1]
```diff
@@ -0,0 +1,181 @@
...
+        [Option('a', "assemblyPath", HelpText = "The path of the assembly.")]
+        public string AssemblyPath { get; set; } = "";
+
+        [Option('o', "outputPath", HelpText = "The directory path of the output results files.")]
+        public string OutputDirectory { get; set; } = "";
+
+        [Option('t', "outputType", HelpText = "The output type for results.")]
+        public ResultsType ResultsType { get; set; } = ResultsType.Unspecified;
+
+        public string OutputFile { get; set; } = "";
```
Comment (baseTwo) [inline Demo/CLI/Helpers/CommandLineOptions.cs]: The setter must be private

File: Demo/CLI/Helpers/CommandLineOptions.cs [hunk 2]
```diff
@@ -0,0 +1,181 @@
...
+
+        [Option('v', "valuesets", HelpText = "The folder where valuesets are stored.")]
+        public string ValueSetsDirectory { get; set; } = "";
+
+        [Option('r', "resourcesDirectory", HelpText = "The directory where the library resources are stored.")]
+        public string ResourcesDirectory { get; set; } = "";
+
+        public string LibraryFile => Path.Combine(ResourcesDirectory, $"{Library}.json");
+
+        public AssemblySource AssemblySource { get; set; } = AssemblySource.Unspecified;
```
Comment (baseTwo) [inline Demo/CLI/Helpers/CommandLineOptions.cs]: The setter must be private

**Review by baseTwo — DISMISSED** [review]
Please have a look through the comments and decide if you'd like to create separate issues for those. I might not be able to immediately review changes to this PR due to illness

**Comment by richfirely** [comment]
> CLI input argument validations should not be displayed as an exception to the end-user. (perhaps another issue for later if you are pressed for time)

Thanks! I am not clear on what you mean by this one. I will fix the rest and merge it for now and revisit that when you are in the office.

**Review by alexzautke — APPROVED** [review]

**Pull request merged**